### ISSUE TITLE
remove oss-fuzz/targets/re2/re2_fuzzer.cc and use re2/blob/master/re2/fuzzing/re2_fuzzer.cc instead

### ISSUE THREAD

**Issue by kcc** [google/oss-fuzz#82]
RE2 repo has its own fuzz target, we should be using that one. 
https://github.com/google/re2/blob/master/re2/fuzzing/re2_fuzzer.cc

**Comment by inferno-chromium** [comment]
@junyer - can you help to maintain re2 fuzzing in OSS-Fuzz. I see you added re2_fuzzer.cc in re2 repo, so this should be a simple fix in build.sh to use that. Also, best to add yourself in project.yaml as well.

**Comment by junyer** [comment]
Sent https://github.com/google/oss-fuzz/pull/136.

**Comment by inferno-chromium** [comment]
@junyer - just a fyi, i was just checking crash stats, re2 fuzzer is hitting a ton of hangs and that might be the reason it is not finding any other bugs. can you please check out https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=175 and fix that so that fuzzer can move forward.

**Comment by junyer** [comment]
Is that still current now that you use the fuzzer distributed with RE2? It has a check for program fanout that should defend against cases like that. (I fixed this for @kcc last year, but the change must not have made it into Chromium's copy of the fuzzer.)

**Comment by inferno-chromium** [comment]
Looks like we need to wait for builds to pick up your change. Once that is done, will comment back if that hang is gone. Or ClusterFuzz will also auto-close https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=175 then.

Right now, builds broken, see tracking - #141

**Comment by junyer** [comment]
> ClusterFuzz has detected this issue as fixed in range 201612052349:201612060800.

Cool.

**Comment by inferno-chromium** [comment]
That one is gone, but there is a new one. You will see a new bug auto-filed soon.

**Comment by junyer** [comment]
https://github.com/google/re2/blob/master/re2/fuzzing/re2_fuzzer.cc#L26 might just need tweaking if the new issue turns out to be the same as the old issue.

**Comment by inferno-chromium** [comment]
Ok bug is filed - https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=265. I don't understand how to tweak fanout value, can you please take a look.

**Comment by junyer** [comment]
https://github.com/google/re2/commit/13fb267cf32c0b2a87ecf1bf72d89dec57ab0e36 reduced the fuzzer program fanout limit to 9. (The program fanout for the regular expression `$(\$*|){918}$` is 10, so the fuzzer timeout should be avoided.)
